src/models/room.ts

```diff
@@ -3461,16 +3532,20 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
 
         // get members that are NOT ourselves and are actually in the room.
         let otherNames: string[] = [];
-        if (this.summaryHeroes) {
+        if (this.heroes) {
             // if we have a summary, the member state events should be in the room state
-            this.summaryHeroes.forEach((userId) => {
+            this.heroes.forEach((hero) => {
                 // filter service members
-                if (excludedUserIds.includes(userId)) {
+                if (excludedUserIds.includes(hero.userId)) {
                     inviteJoinCount--;
                     return;
                 }
-                const member = this.getMember(userId);
-                otherNames.push(member ? member.name : userId);
+                if (hero.displayName) {
+                    otherNames.push(hero.displayName);
+                } else {
+                    const member = this.getMember(hero.userId);
+                    otherNames.push(member ? member.name : hero.userId);
+                }
```

Comment:
could do with a comment here: what is this for?

Reply:
done in 8a26967f5